### PR TITLE
Add data correlation code

### PR DESCRIPTION
**Pull request by Azeajr**
## What
Add analytics data capturing
- session_id
- middleware to track and log API response times

### CODE REVIEW

**Comment by Azeajr** [comment]
@donnahn87
## **Current concerns**
- uploading a ton of images
- huge images
- prevent users from trying to reuse the model they trained
  - I want the flow that uses the model to finish once the collection is uploaded
  - This will enable me to blow away the training and collection images, models, and sessions to save space
  - Will the users we give this to have images on hand to try this out easily

File: frontend/app/components/upload-section.tsx [hunk 1]
```diff
@@ -30,13 +30,21 @@ const style = {
 type UploadSectionProps = {
   title: string;
   apiEndpoint: string;
+  model?: string;
   sendMatchingFiles: (files: File[]) => void;
   setCollectionEndpoint: (endpoint: string) => void;
+  setSesstionId?: (sessionId: string) => void;
```
Comment (donnahn87) [inline frontend/app/components/upload-section.tsx]: Nitpick - `setSessionId` ? Feel free to leave as is

File: frontend/app/components/upload-section.tsx [hunk 2]
```diff
@@ -30,13 +30,21 @@ const style = {
...
 export default function UploadSection(props: UploadSectionProps) {
-  const { title, apiEndpoint, sendMatchingFiles, setCollectionEndpoint } =
-    props;
+  const {
+    title,
+    apiEndpoint,
+    model,
+    sendMatchingFiles,
+    setCollectionEndpoint,
+    setSesstionId,
```
Comment (donnahn87) [inline frontend/app/components/upload-section.tsx]: Nitpick - same as previous comment. Feel free to leave as is

File: frontend/app/components/upload-section.tsx [hunk 3]
```diff
@@ -150,6 +160,7 @@ export default function UploadSection(props: UploadSectionProps) {
       if (isCollection) {
         setScoredFilePaths(responseJson.data);
         handleMatchingFiles(responseJson.data);
+        setSesstionId ? setSesstionId(responseJson.meta.sessionId) : null;
```
Comment (donnahn87) [inline frontend/app/components/upload-section.tsx]: Nitpick - same as previous comment. Feel free to leave as is

File: frontend/app/page.tsx [hunk 4]
```diff
@@ -10,43 +10,84 @@ import "@fontsource/roboto/700.css";
...
         <div className="flex flex-row gap-5">
-          <UploadSection title="Upload Training Images" apiEndpoint={trainingEndpoint} sendMatchingFiles={sendMatchingFiles} setCollectionEndpoint={setCollectionEndpoint} />
-          <UploadSection title="Upload Image Collection" apiEndpoint={collectionEndpoint} sendMatchingFiles={sendMatchingFiles} setCollectionEndpoint={setCollectionEndpoint} />
+          <UploadSection
+            title="Upload Training Images"
+            apiEndpoint={trainingEndpoint}
+            model={model}
+            sendMatchingFiles={sendMatchingFiles}
+            setCollectionEndpoint={setCollectionEndpoint}
+            setSesstionId={applySessionIdToForm}
```
Comment (donnahn87) [inline frontend/app/page.tsx]: Nitpick - same as previous comment. Feel free to leave as is

File: frontend/app/page.tsx [hunk 5]
```diff
@@ -10,43 +10,84 @@ import "@fontsource/roboto/700.css";
...
+          />
+          <UploadSection
+            title="Upload Image Collection"
+            apiEndpoint={collectionEndpoint}
+            model={model}
+            sendMatchingFiles={sendMatchingFiles}
+            setCollectionEndpoint={setCollectionEndpoint}
+            setSesstionId={applySessionIdToForm}
```
Comment (donnahn87) [inline frontend/app/page.tsx]: Nitpick - same as previous comment. Feel free to leave as is

**Review by donnahn87 — DISMISSED** [review]
Overall looks good. Just had nitpick nonbreaking comments for removing the t in `setSesstionId` instances

**Pull request merged**